knowledge-base/dropdownlist-custom-sorting-order-groups.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Custom Sorting Order for Groups in DropDownList
+description: Learn how to apply a custom order to groups in the DropDownList for Blazor.
+type: how-to
+page_title: How to Set a Custom Grouping Order in Blazor DropDownList
+slug: dropdownlist-custom-sorting-order-groups
+tags: dropdownlist, blazor, telerik, grouping, custom order
+res_type: kb
+ticketid: 1666981, 1579415
+---
+
+## Environment
+
+<table>
+<tbody>
+<tr>
+<td>Product</td>
+<td>DropDownList for Blazor</td>
+</tr>
+<tr>
+<td>Version</td>
+<td>6.2.0</td>
+</tr>
+</tbody>
+</table>
+
+## Description
+
+I am using the [grouping feature of the DropDownList]({%slug components/dropdownlist/grouping%}). My DropDownList model has a nested model as property. The DropDownList `GroupField` parameter is bound to a property of the nested model. I want the groups to appear in a non-alphabetical, custom order. How can I customize the grouping order in the DropDownList?
+
+## Solution
+
+To sort the groups in a DropDownList by a custom order, perform a manual sorting operation in the [`OnRead` event]({%slug components/dropdownlist/events%}#onread) handler. Follow these steps:
+
+1. Create a list of strings that represents the values of the group headers. This list will determine the preferred sorting order.
+2. Cast the [`DataSourceResult`]({%slug common-features-data-binding-onread%}#event-argument)  to [`AggregateFunctionsGroup`](/blazor-ui/api/Telerik.DataSource.AggregateFunctionsGroup).
+3. Sort the casted data using the [`Sort` method](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.sort?view=net-8.0) with a custom comparison function.
```

Comment:
This one needs to be an external link. [Source.](https://docs.telerik.com/style-guide/cross-references#guidelines)